### DISCUSSION TITLE
[General] Code example of using wagmi to send EIP-712 tx

### DISCUSSION THREAD

**Discussion by cesarenaldi** [zkSync-Community-Hub/zksync-developers#752] [General] [answered]
### Team or Project

Lens Team

### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

I could only find this example: https://code.zksync.io/tutorials/guide-wagmi#send-transaction

For context, what I am trying to understand is the recommended flow from acquiring access to a WalletClient configured for a ZKsync chain to send the transaction in a wagmi/viem idiomatic way (i.e., leveraging their types system).

**Comment by alhenaai** [comment]
For sending EIP-712 formatted transactions using wagmi with ZKsync, you can follow the general steps outlined in the wagmi guide for ZKsync. Here’s a brief overview of the process:

1. **Connect Wallet**: First, ensure that you have connected to a wallet. You can use the wagmi library to facilitate wallet connections.
   - [Connect Wallet Guide](https://code.zksync.io/tutorials/guide-wagmi#connect-wallet)

2. **Fetch Account**: Once the wallet is connected, fetch the account details.
   - [Fetch Account Guide](https://code.zksync.io/tutorials/guide-wagmi#fetch-account)

3. **Send Transaction**: For sending an EIP-712 transaction, you would typically need to prepare the transaction data according to the EIP-712 standard. This involves setting up the data structure and types for the transaction.
   - [Send Transaction Guide](https://code.zksync.io/tutorials/guide-wagmi#send-transaction)

Unfortunately, the specific example of sending an EIP-712 transaction using wagmi with ZKsync is not directly available in the documentation. However, the general steps for sending transactions can be adapted by incorporating EIP-712 data formatting into the transaction preparation step.

For more detailed guidance and to discuss with other developers, you might find it helpful to visit the ZKsync developers discussions on Github: [ZKsync Developers Discussions](https://github.com/zkSync-Community-Hub/zksync-developers/discussions).

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by bxpana** [reply, marked as answer]
Since wagmi is a wrapper over Viem you can use the Viem docs, https://viem.sh/docs/chains/zksync#utilities.

**Comment by cesarenaldi** [comment]
hey @bxpana, I guess that what I am saying is that, although viem/zksync provides typed actions, I couldn't find a clean path (one that does not involve an unsafe TS assertion) to get a `WalletClient` returned by wagmi hooks to work with the ZKsync specific viem actions.

Hence I am wondering if you had tried this flow.

**Reply by bxpana** [reply]
Sorry about missing this so long ago. You might be able to use this as a reference to help if you didn't already get this figured out. 

https://github.com/matter-labs/zksync-sso/blob/main/examples/nft-quest/composables/useMintNft.ts
